client/network/src/protocol/sync.rs

```diff
@@ -1508,25 +1521,22 @@ impl<B: BlockT> ChainSync<B> {
 			return PollBlockAnnounceValidation::ImportHeader { is_best, announce, who }
 		}
 
-		if number <= self.best_queued_number {
-			trace!(
-				target: "sync",
-				"Added sync target for block announced from {}: {} {:?}",
-				who,
-				hash,
-				announce.header,
-			);
-			self.fork_targets
-				.entry(hash.clone())
-				.or_insert_with(|| ForkTarget {
-					number,
-					parent_hash: Some(*announce.header.parent_hash()),
-					peers: Default::default(),
-				})
-				.peers.insert(who.clone());
-		}
+		trace!(
+			target: "sync",
+			"Added sync target for block announced from {}: {} {:?}",
+			who,
+			hash,
+			announce.summary(),
+		);
+		self.fork_targets
```

Comment:
So, the problem was that we only synced forks if the block number was below our best number?

Reply:
If the block was below best, sync assumed it will be downloaded by simply advancing from peers common number. `fork_targets` was intended for old forks that are behind our current best block, but it is clear now this mechanism should be used for parallel current forks as well.